apps/notifications/src/app/templates/action-dropdown.tsx

```diff
@@ -0,0 +1,46 @@
+import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { moreVertical } from '@wordpress/icons';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getActions } from '../../panel/helpers/notes';
+import { trashNote } from '../../panel/state/notes/thunks';
+import { Note } from '../types';
+
+export default function ActionDropdown( { note, goBack }: { note: Note; goBack: () => void } ) {
+	const dispatch = useDispatch();
+
+	const [ isDeleting, setIsDeleting ] = useState( false );
+
+	const actions = getActions( note );
+	const hasDeleteAction = actions?.hasOwnProperty( 'trash-comment' );
+
+	if ( ! hasDeleteAction ) {
+		return null;
+	}
+
+	const handleDelete = async () => {
+		setIsDeleting( true );
+		await ( dispatch as any )( trashNote( note, true ) );
+		goBack();
+	};
+
+	return (
+		<DropdownMenu icon={ moreVertical } label={ __( 'Actions' ) }>
+			{ ( { onClose } ) => {
+				return (
+					<MenuGroup>
+						<MenuItem
+							onClick={ async () => {
+								await handleDelete();
+								onClose();
+							} }
+						>
+							{ isDeleting ? __( 'Moving to the Trash…' ) : __( 'Trash' ) }
```

Comment:
The copy seems a bit longer 😅

Reply:
"Trashing" isn't a word 😆

Reply:
Or at least, it means something different 🙊

Reply:
I spent a while last night to come up with the right phrase 😭